lib/taurus/core/tango/tangoattribute.py

```diff
@@ -436,21 +436,26 @@ def write(self, value, with_read=True):
             self.error("[Tango] write failed: %s" % str(e))
             raise e
 
-    def poll(self, **kwargs):
+    def poll(self,  single=True, value=None, time=None, error=None):
         """ Notify listeners when the attribute has been polled"""
-        single = kwargs.get('single', True)
         try:
             if single:
                 self.read(cache=False)
             else:
-                value = self.decode(kwargs.get('value'))
-                self.__attr_err = kwargs.get('error')
+                value = self.decode(value)
+                self.__attr_err = error
                 filter_old_event = getattr(tauruscustomsettings,
                                            'FILTER_OLD_TANGO_EVENTS', False)
-                if self.__attr_value is not None and \
-                        self.__attr_err is None and \
-                        filter_old_event and \
-                        kwargs.get('time') < self.__attr_value.time.totime():
+
+                # Discard "valid" notifications (value is not None and error
+                # is None) if FILTER_OLD_TANGO_EVENTS is enabled
+                # and the given timestamp is older than the timestamp
+                # of the cache value
+                if (self.__attr_value is not None
+                    and self.__attr_err is None
```

Comment:
The check of self.__attr_err is None could be avoided if we put this filter after the block of line 517